pilot/pkg/model/sidecar.go

```diff
@@ -602,43 +604,37 @@ func (ilw *IstioEgressListenerWrapper) selectVirtualServices(virtualServices []c
 		if importedHosts, nsFound := hosts[configNamespace]; nsFound {
 			for _, importedHost := range importedHosts {
 				// Check if the hostnames match per usual hostname matching rules
-				hostFound := false
+				if _, ok := vsMap[vsName]; ok {
+					break
+				}
 				for _, h := range rule.Hosts {
-					// TODO: This is a bug. VirtualServices can have many hosts
-					// while the user might be importing only a single host
-					// We need to generate a new VirtualService with just the matched host
+					// VirtualServices can have many hosts, so we need to avoid appending
+					// duplicated virtualservices to slice importedVirtualServices
 					if importedHost.Matches(host.Name(h)) {
 						importedVirtualServices = append(importedVirtualServices, c)
-						hostFound = true
+						vsMap[vsName] = struct{}{}
 						break
 					}
 				}
-
-				if hostFound {
-					break
-				}
 			}
 		}
 
 		// Check if there is an import of form */host or */*
 		if importedHosts, wnsFound := hosts[wildcardNamespace]; wnsFound {
 			for _, importedHost := range importedHosts {
 				// Check if the hostnames match per usual hostname matching rules
-				hostFound := false
+				if _, ok := vsMap[vsName]; ok {
+					break
+				}
 				for _, h := range rule.Hosts {
-					// TODO: This is a bug. VirtualServices can have many hosts
-					// while the user might be importing only a single host
-					// We need to generate a new VirtualService with just the matched host
+					// VirtualServices can have many hosts, so we need to avoid appending
+					// duplicated virtualservices to slice importedVirtualServices
 					if importedHost.Matches(host.Name(h)) {
 						importedVirtualServices = append(importedVirtualServices, c)
-						hostFound = true
+						vsMap[vsName] = struct{}{}
```

Comment:
I donot think this is what the TODO want. This doesnot change any behavior. 

Also we need some unit tests

Reply:
Hi @hzxuzhonghu, Yes, the original TODO want to clone the vs with only one host in field hosts, and I also provided a commit here: https://github.com/istio/istio/pull/33565/commits/d6a528a31004f61b409bbd64ce95295eda013673, however, @ramaraochavali has a suggestion [here](https://github.com/istio/istio/pull/33565#issuecomment-870593964), so I changed the code in this submit. What's your opinion on this PR? BTW, of course, I will add some unit tests, including the VS with multiple hosts.

Reply:
Moreover, I think the change is that there would be no duplicated vs can be appended into `importedVirtualServices`, and less times to do host match operations if one vs has been appended.

Reply:
Trying to understand